jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -102,12 +103,26 @@ private enum RequestState
 
     /*
      * The input readiness state.
+     * <pre>
+     *               read() without preceding isReady()
+     *                         ------
+     *                         \     \   unhandle() returns Action.READ_CALLBACK,
+     *                          \     \  or read() stole available content after setReadListener()
+     *                           ->  IDLE <--------------
+     *     blocking read() unblocked  ^                  \
+     *                                |                   \
+     *                                |                    \  setReadListener called while
+     *            registering demand  v                     v content is available
+     *                               UNREADY ------------> READY
+     *                                         demand
+     *                                         serviced
+     * </pre>
      */
     private enum InputState
     {
-        IDLE,       // No isReady; No data
-        UNREADY,    // isReady()==false; No data
-        READY       // isReady() was false; data is available
+        IDLE,       // No ReadListener notification pending
```

Comment:
Please make full javadocs with the following:
* what state it represents
* when this state is reached
* when this state is left

Reply:
Done.